osu.Game/Online/API/Requests/Responses/APIUser.cs

```diff
@@ -1,14 +1,217 @@
+            set => PlayStyles = value?.Select(str => Enum.Parse(typeof(PlayStyle), str, true)).Cast<PlayStyle>().ToArray();
+        }
+
+        public PlayStyle[] PlayStyles { get; set; }
+
+        [JsonProperty(@"playmode")]
+        public string PlayMode { get; set; }
+
+        [JsonProperty(@"profile_order")]
+        public string[] ProfileOrder { get; set; }
+
+        [JsonProperty(@"kudosu")]
+        public KudosuCount Kudosu { get; set; }
+
+        [JsonProperty(@"statistics")]
+        public UserStatistics Statistics { get; set; }
+
+        [JsonProperty(@"rankHistory")]
+        public RankHistoryData RankHistory
+        {
+            set => Statistics.RankHistory = value;
+        }
+
+        [JsonProperty("badges")]
+        public Badge[] Badges { get; set; }
+
+        [JsonProperty("user_achievements")]
+        public UserAchievement[] Achievements { get; set; }
+
+        [JsonProperty("monthly_playcounts")]
+        public UserHistoryCount[] MonthlyPlaycounts { get; set; }
+
+        [JsonProperty("replays_watched_counts")]
+        public UserHistoryCount[] ReplaysWatchedCounts { get; set; }
+
+        public User ToUser() => new User
+        {
+            Id = Id,
+            JoinDate = JoinDate,
+            Username = Username,
+            PreviousUsernames = PreviousUsernames,
+            Country = Country,
+            Colour = Colour,
+            AvatarUrl = AvatarUrl,
+            CoverUrl = CoverUrl,
+            Cover = Cover,
+            IsAdmin = IsAdmin,
+            IsSupporter = IsSupporter,
+            SupportLevel = SupportLevel,
+            IsGMT = IsGMT,
+            IsQAT = IsQAT,
+            IsBNG = IsBNG,
+            IsBot = IsBot,
+            Active = Active,
+            IsOnline = IsOnline,
+            PMFriendsOnly = PMFriendsOnly,
+            Interests = Interests,
+            Occupation = Occupation,
+            Title = Title,
+            Location = Location,
+            LastVisit = LastVisit,
+            Twitter = Twitter,
+            Lastfm = Lastfm,
+            Skype = Skype,
+            Discord = Discord,
+            Website = Website,
+            PostCount = PostCount,
+            FollowerCount = FollowerCount,
+            FavouriteBeatmapsetCount = FavouriteBeatmapsetCount,
+            GraveyardBeatmapsetCount = GraveyardBeatmapsetCount,
+            LovedBeatmapsetCount = LovedBeatmapsetCount,
+            RankedAndApprovedBeatmapsetCount = RankedAndApprovedBeatmapsetCount,
+            UnrankedBeatmapsetCount = UnrankedBeatmapsetCount,
+            ScoresFirstCount = ScoresFirstCount,
+            PlayStyles = PlayStyles,
+            PlayMode = PlayMode,
+            ProfileOrder = ProfileOrder,
+            Kudosu = Kudosu,
+            Statistics = Statistics,
+            Badges = Badges,
+            Achievements = Achievements,
+            MonthlyPlaycounts = MonthlyPlaycounts,
+            ReplaysWatchedCounts = ReplaysWatchedCounts
+        };
```

Comment:
So this almost 50-line copy constructor shows the whole problem I have with this `User` stuff in general.

Why is *all of it* here? Now I realise setting this right is going to be a toothache and a half, but it's just... Almost *fifty* fields. I guarantee you some usages won't have 80% of this filled out. This is like a C WinAPI structure that you're supposed to fill half of and leave the rest blank for some unknowable reason.

IMO it needs to go. And go *properly*, not by adding another 50-property `APIUser`.